src/diagnostics/riscv64/disasm-riscv64.cc

```diff
@@ -1537,33 +1537,45 @@ void Decoder::DecodeCRType(Instruction* instr) {
 }
 
 void Decoder::DecodeCAType(Instruction* instr) {
-  switch (instr->RvcFunct6Value()) {
-    case 0b100011:
-      switch (instr->RvcFunctValue()) {
-       case 0b00: //c.sub
-         if (instr->RvcRdValue() != 0 && instr->RvcRs2Value() != 0)
-           Format(instr, "sub       'Crs1s, 'Crs1s, 'Crs2s");
-         else
-           UNSUPPORTED_RISCV();
-         break;
-       default:
-         UNSUPPORTED_RISCV();
-      }
+  switch ((C1 | (instr->InstructionBits() & kCATypeMask) |
+          (instr->InstructionBits() & kRvcFunct2Mask))) {
+    case RO_C_SUB:
+      if (instr->RvcRdValue() != 0 && instr->RvcRs2Value() != 0)
```

Comment:
Like mentioned above, this condition could be removed, because the 3-bit reg could be `000` ~ `111`, no invalid value, right?

Besides, the method used here should be `instr->RvcRs1sValue()`, not `instr->RvcRdValue()`, also `instr->RvcRs2sValue()`